src/adapters/mock/adapter.ts

```diff
@@ -26,20 +23,75 @@ export async function mockConvertResponse(
   return Promise.resolve(response);
 }
 
+type RequestListEntry = NormalizedRequest & {attempts: number};
+type ResponseListEntry = NormalizedResponse | Error;
+let requestList: RequestListEntry[] = [];
+let responseList: ResponseListEntry[] = [];
+
+export function getOldestRequest(): NormalizedRequest {
+  if (requestList.length === 0) {
+    throw new Error('No requests have been made');
+  }
+  return requestList.shift()!;
+}
+
+export function getMostRecentRequest(): NormalizedRequest {
+  if (requestList.length === 0) {
+    throw new Error('No requests have been made');
+  }
+  return requestList.pop()!;
+}
```

Comment:
I don't actually use these anywhere, in preference for just `findRequest()` (below).  Not sure if there's a need for them.